src/components/BootProto.js

```diff
@@ -0,0 +1,128 @@
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { Modal, ModalVariant, Button, FormSelect, FormSelectOption, TextInputBase } from '@patternfly/react-core';
+import { Table, TableHeader, TableBody, TableVariant } from '@patternfly/react-table'
+import { useNetworkDispatch } from '../NetworkContext';
+import cockpit from 'cockpit';
+
+const _ = cockpit.gettext;
+
+const BOOT_PROTOCOL_MODES = {
+    none:    _("None"),
+    static:  _("Static"),
+    dhcp:    _("DHCP")
+};
+
+const bootProtocolOptions = Object.keys(BOOT_PROTOCOL_MODES).map(key => {
+    return { value: key, label: BOOT_PROTOCOL_MODES[key] };
+});
+
+const linkText = (connection) => {
+  if (!connection) {
+      return _("Not configured") ;
+  }
+
+  if (connection.iP) {
+    return connection.iP;
+  }
+
+  return BOOT_PROTOCOL_MODES[connection.bootProto];
+};
+
+const BootProto = ({ connection }) => {
+    const [modal, setModal] = useState(false);
+    const [bootProto, setBootProto] = useState(undefined);
+    const [ip, setIp] = useState(connection?.ip || "");
+    const [netmask, setNetmask] = useState(connection?.netmask || "");
+
+    const dispatch = useNetworkDispatch();
+
+    const updateConnection = () => {
+        dispatch({
+            type: 'update_connection',
+            payload: {
+                id: connection.id,
+                changes: { bootProto, ip, netmask }
+            }
+        });
+
+        setModal(false);
+    };
+
+    const renderAddressFields = () => {
+      return (
+          <Table
+              aria-label={_("Static address")}
+              variant={TableVariant.compact}
+              cells={[_("IP"), _("Netmask")]}
+              rows={[
+                {
+                    cells: [
+                        // NOTE: there is a bug with TextInput inside a Table component. The
+                        // workaround is to use TextInputBase instead.
+                        // See https://github.com/patternfly/patternfly-react/issues/4072
+                        <><TextInputBase placeholder={_("IP")} value={ip} onChange={(value) => setIp(value)} id={`connection-${connection?.id}-ip`} /></>,
+                        <><TextInputBase placeholder={_("Netmask")} value={netmask} onChange={(value) => setNetmask(value)} id={`connection-${connection?.id}-netmask`} /></>
+                    ]
+                }
+              ]}
+          >
+              <TableHeader />
+              <TableBody />
+          </Table>
+      );
+    };
+
+    useEffect(() => {
+        setBootProto(connection?.bootProto)
+    }, [connection]);
+
+    return (
+        <>
+            <a href="#" onClick={() => setModal(true)}>{linkText(connection)}</a>
+            <Modal
+                variant={ModalVariant.small}
+                title={_("Boot Protocol")}
```

Comment:
I would call it address as we do actually in yast2-network. At the end we not only modify the boot protocol but we are also modifying the IP Address in case of "Static"

```suggestion
                title={_("Address")}
```